src/lambda/local/debugConfigurationProvider.ts

```diff
@@ -0,0 +1,265 @@
+    readonly preLaunchTask?: string
+    readonly address: 'localhost'
+    readonly port: number
+    readonly localRoot: string
+    readonly remoteRoot: '/var/task'
+    readonly protocol: 'legacy' | 'inspector'
+    readonly skipFiles?: string[]
+}
+
+interface LambdaWithPreLaunchTask {
+    lambda: LocalLambda
+    task: string
+}
+
+interface TasksConfig {
+    version: '2.0.0'
+    tasks?: {
+        label?: string
+    }[]
+}
+
+export class NodeDebugConfigurationProvider implements DebugConfigurationProvider {
+    public async resolveDebugConfiguration(
+        folder: WorkspaceFolder | undefined,
+        debugConfiguration: DebugConfiguration,
+        token?: CancellationToken
+    ): Promise<NodeDebugConfiguration> {
+        throw new Error('Not Implemented')
+    }
+
+    public async provideDebugConfigurations(
+        folder: WorkspaceFolder | undefined,
+        token?: CancellationToken
+    ): Promise<NodeDebugConfiguration[]> {
+        if (!folder) {
+            console.error('Cannot provide debug configuration if no workspace is open.')
+
+            return []
+        }
+
+        const npmProject = await this.findNpmProject(folder, token)
+        // tslint:disable-next-line:no-invalid-template-strings
+        const localRoot = !!npmProject ? join('${workspaceFolder}', npmProject) : '${workspaceFolder}'
+
+        const localLambdas: LambdaWithPreLaunchTask[] = await Promise.all(
+            (await detectLocalLambdas([ folder ])).map(async localLambda => ({
+                lambda: localLambda,
+                task: await this.addPreLaunchTask(folder, localLambda.lambda, {}, 5858)
+            }))
+        )
+
+        return localLambdas.reduce(
+            (accumulator: NodeDebugConfiguration[], localLamdba: LambdaWithPreLaunchTask) => {
+                accumulator.push(
+                    {
+                        type: 'node',
+                        request: 'launch',
+                        name: `Lambda: Debug ${localLamdba.lambda.lambda} locally`,
+                        preLaunchTask: localLamdba.task,
+                        address: 'localhost',
+                        port: 5858,
+                        localRoot,
+                        remoteRoot: '/var/task',
+                        protocol: 'inspector',
+                        skipFiles: [
+                            '/var/runtime/node_modules/**/*.js',
+                            '<node_internals>/**/*.js'
+                        ]
+                    },
+                    {
+                        type: 'node',
+                        request: 'attach',
+                        name: `Lambda: Attach to ${localLamdba.lambda.lambda} locally`,
+                        preLaunchTask: undefined,
+                        address: 'localhost',
+                        port: 5858,
+                        localRoot,
+                        remoteRoot: '/var/task',
+                        protocol: 'inspector',
```

Comment:
This should be intelligently set based on the version of NodeJS: `inspector` for 8.0+, and `legacy` for prior versions.